src/contracts/core/AVSDirectory.sol

```diff
@@ -52,37 +52,89 @@ contract AVSDirectory is
 
     /**
      *
-     *                         EXTERNAL FUNCTIONS
+     *                    EXTERNAL FUNCTIONS
      *
      */
 
     /**
-     * @notice Initializes an operator set for an AVS.
+     * @notice Called by an AVS to create a list of new operatorSets.
      *
-     * @param operatorSetId The ID of the operator set to initialize.
+     * @param operatorSetIds The IDs of the operator set to initialize.
      *
      * @dev msg.sender must be the AVS.
+     * @dev The AVS may create operator sets before it becomes an operator set AVS.
      */
-    function createOperatorSet(uint32 operatorSetId) external {
+    function createOperatorSets(uint32[] calldata operatorSetIds) external {
+        for (uint256 i = 0; i < operatorSetIds.length; ++i) {
+            require(
+                !isOperatorSet[msg.sender][operatorSetIds[i]],
+                "AVSDirectory.createOperatorSet: operator set already exists"
+            );
+            isOperatorSet[msg.sender][operatorSetIds[i]] = true;
+            emit OperatorSetCreated(msg.sender, operatorSetIds[i]);
+        }
+    }
+
+    /**
+     * @notice Sets the AVS as an operator set AVS, preventing legacy M2 operator registrations.
+     *
+     * @dev msg.sender must be the AVS.
+     */
+    function becomeOperatorSetAVS() external {
```

Comment:
do we need this? it should just be upon first `createOperatorSets`

Reply:
This prevents footguns for AVSs. Suggestion by @stevennevins and @wadealexc

Reply:
I was thinking of it as we don't want to let createOperatorSets be called unintentionally. Requiring that this is called first prevents that